packages/pyright-internal/src/languageService/hoverProvider.ts

```diff
@@ -255,37 +260,49 @@ export class HoverProvider {
         sourceMapper: SourceMapper,
         evaluator: TypeEvaluator,
         node: StringNode,
-        type: ClassType,
+        type: Type,
         parts: HoverTextPart[]
     ) {
         // If the expected type is a TypedDict and the current node is a key entry then we can provide a tooltip
         // with the type of the TypedDict key and its docstring, if available.
+        doForEachSubtype(type, (subtype) => {
+            if (isClassInstance(subtype) && ClassType.isTypedDictClass(subtype)) {
+                const entry = subtype.details.typedDictEntries?.get(node.value);
+                if (entry) {
+                    // If we have already added parts for another declaration (e.g. for a union of TypedDicts that share the same key)
+                    // then we need to add a separator to prevent a visual bug.
+                    if (parts.length > 0) {
+                        parts.push({ text: '\n\n---\n' });
+                    }
```

Comment:
Visual bug encountered:
![bug](https://user-images.githubusercontent.com/23125036/163497549-9d24f4db-3b89-45e4-a22e-d2c3ed84d51b.jpeg)

Reply:
This fix is probably in the wrong place but I didn't want to make a potentially mass breaking change.